src/components/TextArea/__tests__/TextArea.spec.tsx

```diff
@@ -1,33 +1,54 @@
 import React from 'react';
-import toJson from 'enzyme-to-json';
+import { render, RenderResult, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
 import { TextArea } from '../TextArea';
 
 describe('<TextArea> that renders a textarea', () => {
-    const data = {
-        target: {
-            value: 'test',
-        },
-    };
+    const data = 'test';
+
+    let view: RenderResult;
 
     it('should render default textarea correctly', () => {
-        const wrapper = shallow(<TextArea defaultValue="Some value" />);
-        expect(toJson(wrapper)).toMatchSnapshot();
+        view = render(<TextArea defaultValue="Some value" />);
+
+        expect(view.container).toMatchSnapshot();
+        expect(screen.getByRole('textbox')).toBeInTheDocument();
     });
 
     it('should add classes when props are changed', () => {
-        const wrapper = shallow(<TextArea size="large" isBlock disabled />);
-        expect(toJson(wrapper)).toMatchSnapshot();
+        view = render(<TextArea size="large" isBlock disabled />);
+
+        const textarea = screen.getByRole('textbox');
+
+        expect(view.container).toMatchSnapshot();
+        expect(textarea).toBeInTheDocument();
+        expect(textarea).toHaveAttribute('disabled');
+        expect(textarea).toHaveClass('TextArea TextArea--size_large TextArea--isBlock');
     });
 
-    it('should call change callback correctly', () => {
+    it.skip('should call change callback correctly', async () => {
+        const user = userEvent.setup();
         const onChange = jest.fn();
-        const wrapper = shallow(<TextArea onChange={onChange} />);
-        wrapper.find('textarea').simulate('change', data);
+        view = render(<TextArea onChange={onChange} disabled />);
+
+        expect(view.container).toMatchSnapshot();
+
+        const textarea = screen.getByRole('textbox');
+
+        expect(textarea).toBeInTheDocument();
+
+        await user.type(textarea, data);
```

Comment:
I don't know why but value doesn't change could anyone please check it ?

Reply:
maybe because the TextAre is disabled, so the user cannot type into it?

Reply:
yes, sure. I removed disabled but when call   await user.type(textarea, data); onChange function called 4 times

Reply:
If `data` has 4 letters, then it makes sense. You hit the keyboard 4 times, and it will call `onChange` for each of those

Reply:
I don't get why it returns
![image](https://github.com/textkernel/oneui/assets/104063210/7e00a0ee-7ebf-4ac5-91e7-ea7ee7ffc1ab)

Reply:
1. we expect 4 calls here
2. `toHaveBeenCalledWith` doesn't support multiple calls as far as I remember. 
3. when I try to check `toLastHaveBeenCalledWith('t')` I get the following output:

```
    Expected: "t"
    Received
           3: {"_dispatchInstances": null, "_dispatchListeners": null, "_targetInst": null, "bubbles": null, "cancelable": null, "currentTarget": [Function currentTarget], "defaultPrevented": null, "dispatchConfig": null, "eventPhase": null, "isDefaultPrevented": [Function functionThatReturnsFalse], "isPropagationStopped": [Function functionThatReturnsFalse], "isTrusted": null, "nativeEvent": null, "target": null, "timeStamp": [Function timeStamp], "type": null}
    ->     4: {"_dispatchInstances": null, "_dispatchListeners": null, "_targetInst": null, "bubbles": null, "cancelable": null, "currentTarget": [Function currentTarget], "defaultPrevented": null, "dispatchConfig": null, "eventPhase": null, "isDefaultPrevented": [Function functionThatReturnsFalse], "isPropagationStopped": [Function functionThatReturnsFalse], "isTrusted": null, "nativeEvent": null, "target": null, "timeStamp": [Function timeStamp], "type": null}
```

This is expected as it is the native implementation and it calls onChange with the whole event object. However you can see that the target is still null, which is not what we would expect to happen. 
But the console comes to our aid with it, as with this code you can see a lot of errors when running this test saying: 
```
console.error
    Warning: This synthetic event is reused for performance reasons. If you're seeing this, you're accessing the property `type` on a released/nullified synthetic event. This is set to null. If you must keep the original synthetic event around, use event.persist(). See https://fb.me/react-event-pooling for more information.
```

So we can consent with just testing 
`expect(onChange).toHaveBeenCalledTimes(data.length);`

If you want to, you can go farther and see how to use the `event.persist()` as indicated in the error.


Please let me know if this type of reasoning is helpful for you, or not interesting. Thanks

Reply:
I tried this one 
![image](https://github.com/textkernel/oneui/assets/104063210/21276b87-ec2f-4770-a1e6-8927498bd5d6)
But it doesn't work. I'll use this approach  expect(onChange).toHaveBeenCalledTimes(data.length);